src/components/GradeItemAccordion.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import TextInput from './TextInput';
+import DelButton from './MainActionButton';
+import { CourseItem, Fraction } from '../api';
+export interface GradeItemAccordionInterface {
+	item: CourseItem;
+	updateItem: (item: CourseItem) => void;
+}
+
+const Accordion = styled.tr`
+	background-color: #ffffff;
+	color: #4f4f4f;
+	padding: 2em;
+	justify-content: center;
+	line-height: 8em;
+	transition: line-height 14em ease-out;
+	border: #ffffff;
+	text-decoration: none;
+	border: none;
+`;
+const RowComponent = styled.td`
+	border: #ffffff;
+	font-weight: bold;
+	font-size: large;
+	border: none;
+	margin: 1.5em;
+`;
+const DropDownArrow = styled.button`
+	border: none;
+	outline: none;
+	background-color: #ffffff;
+	font-size: 2em;
+`;
+const AccordionExtended = styled.tr`
+	font-weight: bold;
+	font-size: large;
+	color: #4961e1;
+	border: none;
+`;
+const AccordionExtendedComponent = styled.td`
+	background-color: #ffffff;
+	border: none;
+`;
+
+const DeleteButton = styled(DelButton)`
+	background-color: #ffffff;
+	color: #b80f0a;
+	margin-right: 2em;
+`;
+
+function submit() {
+	console.log('changed');
+}
+
+function gradeToString(grade: number | Fraction | undefined) {
+	if (grade == undefined) {
+		return '';
+	} else if (typeof grade === 'object') {
+		return `${grade.numerator}/${grade.denominator}`;
+	}
+	return `${grade}`;
+}
+
+function parseGrade(s: string) {
+	const fractionRegex = /([\d]*\.?[\d]+)\/([\d]*\.?[\d]+)/;
+	const match = s.match(fractionRegex);
+	console.log({ parseGradeMatch: match });
+	if (!isNaN(Number(s))) {
+		return Number(s);
+	} else if (match) {
+		return { numerator: Number(match[1]), denominator: Number(match[2]) };
+	} else {
+		return undefined;
+	}
+}
+
+function GradeItemAccordion({ item, updateItem }: GradeItemAccordionInterface) {
+	const { name, weight, grade, dueDate } = item;
+	// useEffect(() => {
+	// 	if (grade != tempGrade) setTempGrade(grade);
+	// }, [item]);
+	const [expanded, setExpanded] = useState(false);
+	const [tempName, setTempName] = useState(name);
+	const [tempWeight, setTempWeight] = useState(weight?.toString() || '');
```

Comment:
In theory we could use the gradeToString function here, but this is probably okay